action/action-util/src/main/java/com/powsybl/action/util/GeneratorModificationTask.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.action.util;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.contingency.tasks.ModificationTask;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Network;
+
+import java.util.Objects;
+
+/**
+ * @author Olivier Perrin <olivier.perrin at rte-france.com>
+ */
+public class GeneratorModificationTask implements ModificationTask {
+
+    private final String generatorId;
+    private final Modifs modifs;
+
+    public GeneratorModificationTask(String generatorId, Modifs modifs) {
+        this.generatorId = Objects.requireNonNull(generatorId);
+        this.modifs = modifs;
+    }
+
+    @Override
+    public void modify(Network network, ComputationManager computationManager) {
+        Generator g = network.getGenerator(generatorId);
+        if (g == null) {
+            throw new PowsyblException("Generator '" + generatorId + "' not found");
+        }
+        if (modifs.getMinP() != null) {
+            g.setMinP(modifs.getMinP());
+        }
+        if (modifs.getMaxP() != null) {
+            g.setMaxP(modifs.getMaxP());
+        }
+
+        if (modifs.getVoltageRegulatorOn() != null) {
+            g.setVoltageRegulatorOn(modifs.getVoltageRegulatorOn());
+        }
+
+        boolean skipOtherConnectionChange = false;
```

Comment:
Same kind of remark than @sylvlecl. It is not because  generator is  disconnected that we should not be able to set targetP.

Reply:
When ``targetP`` is provided in a ``generatorModification`` task, we want to connect the generator if it isn't already in a connected state. But if in a same task we specify ``connected false`` and a ``targetP`` value, we want the final generator state to be disconnected.
The ``skipOtherConnectionChange`` is to ensure that the ``targetP`` modification will not reconnect the generator if we disconnected it in the same task.